petastorm/unischema.py

```diff
@@ -390,8 +390,9 @@ def dict_to_spark_row(unischema, row_dict):
                 encoded_dict[field_name] = value.tolist()
             else:
                 encoded_dict[field_name] = value
-
-    return pyspark.Row(**encoded_dict)
+    schema_field_indices = {field_name: i for i, field_name in enumerate(unischema.fields)}
+    sorted_dict = OrderedDict(sorted(encoded_dict.items(), key=lambda item: schema_field_indices[item[0]]))
+    return pyspark.Row(**sorted_dict)
```

Comment:
Sorted_dict still cannot guarantee the result correct. Python<3.6 does not guarantee the kwargs key order.
See https://issues.apache.org/jira/browse/SPARK-29748

Replace the 3 lines with:
~~~python
# generate a value list which match the schema column order.
field_list = [field.name for field in unischema.fields]
value_list = [encoded_dict[field.name] for field in unischema.fields]
# create a row by value list
row = pyspark.Row(value_list)
# set row fields
row.__fields__ = field_list
return row
~~~

CC @mengxr